src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs

```diff
@@ -58,6 +72,7 @@ public FunctionsSyncManager(IConfiguration configuration, IHostIdProvider hostId
             _environment = environment;
             _hostNameProvider = hostNameProvider;
             _functionMetadataProvider = functionMetadataProvider;
+            _extensionsManager = extensionsManager;
```

Comment:
I'm not sure promoting the pattern of having the sync payload be version dependent is a good thing. Currently, the payload for all extensions other than Durable just reflects the actual user specified trigger data from function.json, without any complicated logic.

I think the ScaleController (and runtime scale logic) should be responsible for defaulting any values that aren't present - basically consuming the data as is.

Reply:
I think I agree with you that the ScaleController should be the one responsible for defaulting values. Unfortunately, the ScaleController has no knowledge what version of the durable extension the customer is using, so it cannot properly make that decision with the information it has today. That leaves us with two options.

1. The approach I am taking in this PR, where we decide the default value on the host and send that along as the payload. This has the downside of duplicating our defaulting logic on each version of the Functions host, which is definitely not ideal.
2. We modify the sync triggers API to include what extension version of Durable we are using. That way the ScaleController can make the defaulting decision itself, and we can reduce the complexity on building the sync trigger payload.

Either way, we do need information from extensions manager here. 

In my latest revision, I have included both approach 1 and approach 2, with the idea that approach 1 can be used as a stopgap until we update the ScaleController. After that, I can remove the defaulting logic from the host. Thoughts?

Reply:
You're getting integration test failures because I think the service dependency you added to FunctionsSyncManager is invalid, due to service scopes for the chain of services involved. Some are app level, some host level, causing issues. So it's not just a test issue - if you F5 the host locally and try to hit endpoints, you'll see the same failure.

I understand we need to figure out a tactical fix for now, but we eventually need to get to the point where there is no Durable specific logic in here at all. The host shouldn't have any hard coded extension specific logic in it. Currently only durable is doing this.